src/server/rpc/procedures/waterwatch/waterwatch.js

```diff
@@ -0,0 +1,199 @@
+// TODO change console logs.
+// eslint
+let WaterWatchRPC = {
+    isStateless: true,
+    getPath: () => '/waterwatch'
+};
+
+let debug = require('debug'),
+    // request = require('request'),
+    rp = require('request-promise'), //maybe use request-promise-native?
+    error = debug('netsblox:rpc:trends:error'),
+    CacheManager = require('cache-manager'),
+    trace = debug('netsblox:rpc:trends:trace'),
+    Q = require('q');
+
+let baseUrl = 'https://waterservices.usgs.gov/nwis/iv/?',
+    cache = CacheManager.caching({store: 'memory', max: 1000, ttl: 36000});
+
+
+
+// let northernLat = 38;
+// let easternLong = -81;
+// let southernLat = 37;
+// let westernLong = -82;
+
+
+// turn an options object into query friendly string
+function encodeQueryData(options) {
+   let ret = [];
+   for (let d in options)
+     ret.push(encodeURIComponent(d) + '=' + encodeURIComponent(options[d]));
+   return ret.join('&');
+}
+
+// used to send feedback about errors to the caller.
+function showError(err, response) {
+    error('error',err);
+    response.send('Error occured. Bounding box too big?');
+    // response.json(err);
+}
+
+//sanitize coordinate inputs
+
+//******************************
+// factor out message sending and handling of multiple users = sendNext and Stop
+// msgs: an array of msgs stored in global scope.
+let DELAY = 250;
+let msgs = [];
+
+function sendNextMsg(socket){
+    // QUESTION is there a need to check for the socket ID? no need.
+    if (msgs.length < 1){
+        return;
+    }
+    socket.send(msgs.shift());
+    setTimeout(sendNextMsg,DELAY,socket);
+}
+
+//checks msg ownership, needs access to socket.
+function isMsgOwner(msg){
+    return msg.dstId = socket.roleId
+}
+
+function stopSendingMsgs($this){
+    socket = $this.socket;
+    response = $this.response;
+    if (msgs) {
+        // remove those with a different roleId | dont remove other's messages
+        msgs = msgs.filter(msg => {
+            return msg.dstId != socket.roleId;
+        });
+    }
+    trace(`Stopped sending messages to user with roleId ${socket.roleId}`);
+    response.send('Stopped sending messages.');
```

Comment:
Since we are using the same stop functionality in other RPCs I had this included in this function make it more reusable .